spec/forms/openid_connect_authorize_form_spec.rb

```diff
@@ -34,64 +34,22 @@
   let(:code_challenge_method) { nil }
 
   describe '#submit' do
-    let(:user) { create(:user) }
-    let(:rails_session_id) { SecureRandom.hex }
-    subject(:result) { form.submit(user, rails_session_id) }
+    subject(:result) { form.submit }
 
     context 'with valid params' do
       it 'is successful' do
         allow(FormResponse).to receive(:new)
 
-        form.submit(user, rails_session_id)
-
-        identity = user.identities.where(service_provider: client_id).first
+        form.submit
 
         extra_attributes = {
           client_id: client_id,
-          redirect_uri: "#{redirect_uri}?code=#{identity.session_uuid}&state=#{state}",
+          redirect_uri: nil,
```

Comment:
It should have a key. The `redirect_uri` method didn't change. The value depends on `success`. If it's `true` (i.e the request was valid), then the `redirect_uri` value is the return value of `success_redirect_uri` (which is `nil` in this case because the identity hasn't been linked yet), otherwise of `error_redirect_uri`.